crates/factors-executor/src/lib.rs

```diff
@@ -211,6 +211,16 @@ impl<'a, T: RuntimeFactors, U> FactorsInstanceBuilder<'a, T, U> {
     pub fn factor_builder<F: Factor>(&mut self) -> Option<&mut F::InstanceBuilder> {
         self.factor_builders().for_factor::<F>()
     }
+
+    /// Returns the underlying wasmtime engine for the instance.
+    pub fn engine(&self) -> &spin_core::wasmtime::Engine {
```

Comment:
Slightly confusing because in this same crate `FactorsExecutorApp::engine()` returns a `spin_core::Engine`. Maybe:

```suggestion
    pub fn wasmtime_engine(&self) -> &spin_core::wasmtime::Engine {
```

Reply:
I agree. The change seems like a fine one to make. While we're here though - a larger put perhaps cleaner change might be to rename `spin_core::Engine` to `spin_core::WasmRuntime`. This might be preferrable since even with your suggested change, a user still has to know how to disambiguate when a type signature includes an unqualified `Engine`.

Reply:
or maybe `SpinEngine`? "Runtime" is also awfully overloaded...

Reply:
yet another alternative would be to re-export `wasmtime::Engine` as `spin_core::WasmtimeEngine`